### PR TITLE
Migrate recipient-lists.apib

### PR DESCRIPTION
**Pull request by avigoldman**

### CODE REVIEW

File: content/api/recipient-lists.apib [hunk 1]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
-|name    |string       |Short, pretty/readable recipient list display name, not required to be unique |no  |If a name is not specified, then defaults to the same value as id.  Maximum length - 64 bytes |
-|description |string |Detailed description of the recipient list|no | Maximum length - 1024 bytes|
-|attributes |JSON object  |Recipient list attribute object  |no  |This JSON object allows users to store arbitrary metadata related to this list.  This data is not used by the API.  It is only for the user. |
-|recipients |JSON array  |Array of recipient objects |yes |For a full description, see the [Recipient Attributes](#header-recipient-attributes). |
-
-### Recipient Attributes
+### Recipient List Object
+
++ Data Structure: Attributes
+    + id (string) - Short, unique, identifier. If an id is not specified, one is generated. <br /> Maximum length: 64 bytes
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: Just "Unique  alphanumeric ID", leave the "If an id is .." to the POST description

Comment (jgzamora) [inline content/api/recipient-lists.apib]: Just "Unique alphanumeric ID", leave the "If an id is .." to the POST description

File: content/api/recipient-lists.apib [hunk 2]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
+    + name (string) - Recipient list display name. If a name is not specified, it defaults to the same value as id. <br />  Maximum length: 64 bytes
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: Remove "If name not specified..." Leave it in POST description.

File: content/api/recipient-lists.apib [hunk 3]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
+    + description (string) - Detailed description of the recipient list. <br /> Maximum length: 1024 bytes
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: I've been removing "Detailed" from all of these

File: content/api/recipient-lists.apib [hunk 4]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
+        },
+        "total_accepted_recipients": 3,
+        "recipients": [ ... ]
+      }
+    }
+    ```
+
+### Recipient Object
 
 Recipients are described in a JSON array with the following fields:
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: "Recipients are described in an array of objects with the following attributes:"

File: content/api/recipient-lists.apib [hunk 5]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
-|substitution_data | JSON object | Key/value pairs associated with a recipient that are provided to the substitution engine |no | Recipient substitution data takes precedence over transmission substitution data.  Unlike metadata, substitution data is not included in Webhook events.|
++ Data Structure: Attributes
+    + address (enum, required) - Address information for a recipient. See the [Address Object](#header-address-object).
+        + (object)
+        + (string)
+    + return_path (string) 
+        Email address to use for envelope FROM. Overrides transmission level return_path. <br /> <br /> For <span class="label label-primary"><strong>SparkPost</strong></span> accounts, the domain part of the return_path address must be a [CNAME-verified sending domain](/api/sending-domains/#sending-domains-verify-post).  The local part of the return_path address will be overwritten by SparkPost servers.<br><br>For <a href="https://www.sparkpost.com/enterprise-email/"><span class="label label-warning"><strong>Enterprise</strong></span></a> accounts, the return_path may be any valid email address and the localpart in the return_path will **not** be overwritten by SparkPost servers.
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: Let's get rid of the "For SparkPost " and leave that as the default behavior/description. The Enterprise description then shines as different / won't be missed

File: content/api/recipient-lists.apib [hunk 6]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
+            "job": "Firefighter"
+        }
+    }
+    ```
+    
 
-#### Address Attributes
+### Address Object
 If the `address` field is a string type, it is interpreted as the email address. If it is a JSON
 object, it is described with the following fields:
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: "If the `address` attribute is a string, it is interpreted as the email address. If it is an
 object, it is described with the following attributes:

File: content/api/recipient-lists.apib [hunk 7]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
-|email    |string       |Valid email address   |required if channel is `email` | |
-|name |string |User-friendly name for the email address |no |Used when channel is `email`|
-|header_to|string       |Email address to display in the `To` header instead of _address.email_ (for BCC)|no|Used when channel is `email`|
-|token|string| See [Push Specific Attributes](#header-push-specific-attributes) |required if channel is `gcm` or `apns`|<a href="https://www.sparkpost.com/enterprise-email/"><span class="label label-warning"><strong>Enterprise</strong></span></a> |
-|app_id|string|See [Push Specific Attributes](#header-push-specific-attributes) |required if channel is `gcm` or `apns`| <a href="https://www.sparkpost.com/enterprise-email/"><span class="label label-warning"><strong>Enterprise</strong></span></a> |
-
-#### Notes on channel
-Communication channels other than email are currently only supported for inline recipient lists. Fields unrelated to the value of _channel_ are ignored. A field is considered unrelated if it is not required for that value of _channel_ unless mentioned otherwise in Notes
++ Data Structure
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: no title is going to go to Request Body

File: content/api/recipient-lists.apib [hunk 8]
```diff
@@ -11,52 +11,81 @@ transmission data, or a stored recipient list id attribute can be specified.
...
+    + email (string, required) - Valid email address
+    + name (string) - User-friendly name for the email address
+    + header_to (string)
+        Email address to display in the `To` header instead of _address.email_ ([for CC and BCC](https://support.sparkpost.com/customer/en/portal/articles/2432290-using-cc-and-bcc-with-the-rest-api)) 
 
 **Constructing Headers using the Address Attributes**
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: Clear instances of "JSON" pls

File: content/api/recipient-lists.apib [hunk 9]
```diff
@@ -355,78 +373,27 @@ retrieve the recipients contained in a list, the list must be specified and the
...
-                        "description": "An email list of undergraduate students at UMBC",
-                        "attributes": {
-                            "internal_id": 111,
-                            "list_group_id": 11321
-                        },
-                        "total_accepted_recipients": 8
-                    }
-                ]
-            }
+If a `recipients` array is provided in the update request, it will replace the current recipients. This means that if you're adding, updating, or removing a single recipient, the `recipients` array must contain all current recipients, regardless they are being changed or not.
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: I'd move this one above the structure

File: content/api/recipient-lists.apib [hunk 10]
```diff
@@ -584,21 +550,20 @@ number of rejected recipients will only be returned if a `recipients` array is p
             }
 
 
-## Delete [/recipient-lists/{id}]
+### Delete a Recipient List [DELETE /recipient-lists/{id}]
+
 
-### Delete a Recipient List [DELETE]
 
 Permanently delete the specified recipient list.
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: ❌

File: content/api/recipient-lists.apib [hunk 11]
```diff
@@ -584,21 +550,20 @@ number of rejected recipients will only be returned if a `recipients` array is p
...
 
++ Parameter
+    + id (required, string, `unique_id_4_graduate_students_list`)
+
 Once a recipient list is deleted, it
-cannot be recovered.  Before deleting a list, ensure that it is no longer needed and keep a backup copy.  If a deleted
-list is needed again, the list must be resubmitted with the CREATE API.
+cannot be recovered.  Before deleting a list, ensure that it is no longer needed and keep a backup copy.
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: This isn't rendered anywhere that I can see. Don't think it's necessary either

File: content/api/recipient-lists.apib [hunk 12]
```diff
@@ -584,21 +550,20 @@ number of rejected recipients will only be returned if a `recipients` array is p
...
 
 <banner status="warning">If a transmission contains a recipient list, the recipient list cannot be deleted if the transmission is submitted or generating.</banner>
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: this also isn't rendering for me

File: content/api/recipient-lists.apib [hunk 13]
```diff
@@ -584,21 +550,20 @@ number of rejected recipients will only be returned if a `recipients` array is p
...
 
-+ Parameters
-    + id (required, string, `unique_id_4_graduate_students_list`) ... Identifier of the recipient list
-
 + Request
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: I think the 404 here isn't unnecessary

File: content/api/recipient-lists.apib [hunk 14]
```diff
@@ -650,3 +615,45 @@ list is needed again, the list must be resubmitted with the CREATE API.
                 }
             ]
         }
+
+
+### List all Recipient Lists [GET /recipient-lists]
+
+List a summary of all recipient lists. The recipients for each list are not included in the
```
Comment (jgzamora) [inline content/api/recipient-lists.apib]: "Returns an array abbreviated recipient list objects. The recipients for each list are not included in the results. Retrieve a single list to get a list the recipients. (link this one to the method)"

File: content/api/recipient-lists.apib (hunk 8 above)
Comment (avigoldman) [reply content/api/recipient-lists.apib]: lol sorry

File: content/api/recipient-lists.apib (hunk 12 above)
Comment (avigoldman) [reply content/api/recipient-lists.apib]: Had it in the wrong order cuz of unrelated testing :headdesk:

**Pull request merged**